kratos/includes/table_accessor.h

```diff
@@ -0,0 +1,175 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//                   Riccardo Rossi
+//                   Carlos Roig
+//                   Ruben Zorrilla
+//
+
+# pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/accessor.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class TableAccessor
+ * @ingroup Kratos Core
+ * @brief This class defines the way a certain property is accessed according to a table.
+ * @brief The tables are supposed to relate double <-> double type entities. 
+ * @brief The input variable is suposed to be a nodally accesible one (either historical or not)
+ * @author Alejandro Cornejo, Riccardo Rossi, Carlos Roig and Ruben Zorrilla
+ */
+class KRATOS_API(KRATOS_CORE) TableAccessor : public Accessor
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Geometry type definition
+    using GeometryType = Geometry<Node>;
+
+    /// Variable type
+    using VariableType = Variable<double>;
+
+    /// BaseType
+    using BaseType = Accessor;
+
+    using DataLocation = Globals::DataLocation;
+
+    using SizeType = std::size_t;
+
+    /// Pointer definition of TableAccessor
+    KRATOS_CLASS_POINTER_DEFINITION(TableAccessor);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Custom constructor
+    TableAccessor(VariableType* pInputVariable, const std::string& rInputVariableType = "nodal_historical") 
+        : mpInputVariable(pInputVariable)
+    {
+        // We initialize the variable type only once
+        if (rInputVariableType == "nodal_historical") {
+            mInputVariableType = Globals::DataLocation::NodeHistorical;
+        } else if (rInputVariableType == "nodal_non_historical") {
+            mInputVariableType = Globals::DataLocation::NodeNonHistorical;
+        } else if (rInputVariableType == "elemental_non_historical") {
+            mInputVariableType = Globals::DataLocation::Element;
```

Comment:
```suggestion
        if (rInputVariableType == "node_historical") {
            mInputVariableType = Globals::DataLocation::NodeHistorical;
        } else if (rInputVariableType == "node_non_historical") {
            mInputVariableType = Globals::DataLocation::NodeNonHistorical;
        } else if (rInputVariableType == "element") {
            mInputVariableType = Globals::DataLocation::Element;
```

can you please make the names consistent? I would like to unify this in the near future, i.e. the conversion btw enum and string

See also here: https://github.com/KratosMultiphysics/Kratos/blob/5db44d111c5fdb9935da205abf49760c0f80e6f7/applications/CoSimulationApplication/python_scripts/coupling_interface_data.py#L55